src/node/interfaces.cpp

```diff
@@ -408,7 +409,7 @@ class NodeImpl : public Node
     {
         return MakeSignalHandler(::uiInterface.NotifyBlockTip_connect([fn, this](SynchronizationState sync_state, const CBlockIndex* block) {
             fn(sync_state, BlockTip{block->nHeight, block->GetBlockTime(), block->GetBlockHash()},
-               chainman().GuessVerificationProgress(block));
+                WITH_LOCK(chainman().GetMutex(), return chainman().GuessVerificationProgress(block)));
```

Comment:
nit for when you next have to update

```suggestion
               WITH_LOCK(chainman().GetMutex(), return chainman().GuessVerificationProgress(block)));
```